akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -146,26 +146,25 @@ trait FileAndResourceDirectives {
   def listDirectoryContents(directories: String*)(implicit renderer: DirectoryRenderer): Route =
     get {
       extractRequestContext { ctx ⇒
-        val path = ctx.unmatchedPath
-        val fullPath = ctx.request.uri.path.toString
-        val matchedLength = fullPath.lastIndexOf(path.toString)
-        require(matchedLength >= 0, s"Unmatched path '$path' wasn't a suffix of full path '$fullPath'. " +
-          "This usually means that ctx.unmatchedPath was manipulated inconsistently " +
-          "with ctx.request.uri.path")
-        val pathPrefix = fullPath.substring(0, matchedLength)
-        val pathString = withTrailingSlash(safeJoinPaths("/", path, ctx.log, '/'))
-        val dirs = directories flatMap { dir ⇒
-          safeDirectoryChildPath(withTrailingSlash(dir), path, ctx.log) match {
```

Comment:
Agreed. I think @rbudzko is on Windows but I'll double check. The behavior shouldn't have change here as it's just a refactoring (famous last words...)